components/script/dom/file.rs

```diff
@@ -45,6 +47,19 @@ impl File {
                            FileBinding::Wrap)
     }
 
+    // Construct from selected file message from file manager thread
+    pub fn new_from_selected(window: &Window, selected: SelectedFile) -> Root<File> {
+        let name = DOMString::from(selected.filename.to_str().expect("File name encoding error"));
+
+        // FIXME: fix this after PR #11221 is landed
+        let id = selected.id;
+        let slice = DataSlice::empty();
+
+        let global = GlobalRef::Window(window);
+        // XXX: as i64? looks like DOM codegen and std::fs don't agree on this type
```

Comment:
Windows: https://msdn.microsoft.com/en-us/library/windows/desktop/ms724284(v=vs.85).aspx
Linux: http://stackoverflow.com/questions/471248/what-is-ultimately-a-time-t-typedef-to

Ok, maybe we should left this as it is :(